caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,12 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+// calculateEtag produces a strong etag by default. Prefix the result with "W/" to convert this into a weak one.
+// see https://tools.ietf.org/html/rfc7232#section-2.3
+func calculateEtag(d os.FileInfo) string {
+	return fmt.Sprintf(`"%x-%x"`, d.ModTime().Unix(), d.Size())
```

Comment:
> But the bullish high-performance argument is that timestamp+length is good enough in all normal use-cases.

Because if the contents of the file change, it is highly likely that the timestamp or length will change. I agree that's reasonable in pretty much every case I can think of.

> Thinking about this a little more, the same argument suggests that the compressed artefacts should also have strong etags (based on timestamp+length).

Sure, I don't see why the caching on those needs to be treated any differently? It's still a static file.

Let's try strong E-Tags then.